src/wp-admin/user-edit.php

```diff
@@ -364,12 +367,12 @@
 
 							wp_dropdown_languages(
 								array(
-									'name'      => 'locale',
-									'id'        => 'locale',
-									'selected'  => $user_locale,
-									'languages' => $languages,
-									'show_available_translations' => false,
-									'show_option_site_default' => true,
+									'name'                        => 'locale',
+									'id'                          => 'locale',
+									'selected'                    => $user_locale,
+									'languages'                   => $languages,
+									'show_available_translations' => current_user_can( 'install_languages' ) && wp_can_install_language_pack(),
```

Comment:
```suggestion
									'show_available_translations' => current_user_can( 'install_languages' ),
```

It's a meta capability that checks `wp_can_install_language_pack()` already. See https://core.trac.wordpress.org/changeset/41268